### PR TITLE
better gh workflows

### PR DESCRIPTION
**Pull request by moonso**
This PR updates the GitHub Actions workflows. 
It was confusing when mixing when different workflows should be executed (`push`, `pull_request`). Now everything will run for every push and opened PR.

Black is now allowed to edit files, if this PR is merged then all files will automatically be blacked by the bot in every PR. You might see that in the "files changed" :)

### CODE REVIEW

**Comment by moonso** [comment]
Could someone have a quick look at this @barrystokman or @patrikgrenfeldt ?

**Comment by barrystokman** [comment]
> Could someone have a quick look at this @barrystokman or @patrikgrenfeldt ?

I'll have a look

**Comment by moonso** [comment]
I wonder if it blacked everything again since we use line length of 100 in the action...

File: .github/workflows/lint_and_install.yml [hunk 1]
```diff
@@ -0,0 +1,123 @@
...
+      # [0|0|436] [mans.magnusson@hasta:~] [P_main] 3s $ date
+      # Wed May 27 16:44:09 CEST 2020
+      # [0|0|436] [mans.magnusson@hasta:~] [P_main] $ python --version
+      # Python 3.7.1
+      # (prod170926) [hiseq.clinical@hippocampus ~]$ date
+      # Wed May 27 16:44:09 CEST 2020
+      # (prod170926) [hiseq.clinical@hippocampus ~]$ python --version
+      # Python 3.6.2 :: Continuum Analytics, Inc.
+      # (prod170926) [hiseq.clinical@hippocampus ~]$ pip --version
+      # pip 9.0.1
```
Comment (barrystokman) [inline .github/workflows/lint_and_install.yml:64]: Can we store this information somewhere instead of in comments here?

Comment (moonso) [reply .github/workflows/lint_and_install.yml:64]: I was also thinking about it but could not get any idea where... any suggestion? I think it is good to show why we are choosing these specific versions and where to update information when environments are updated

Comment (barrystokman) [reply .github/workflows/lint_and_install.yml:64]: Maybe a markdown file in `cg/.github/workflows`?

Comment (moonso) [reply .github/workflows/lint_and_install.yml:64]: Good idea! I added that

**Comment by moonso** [comment]
I'll merge and deploy then 👍

**Comment by moonso** [comment]
Skärmavbild 2020-05-28 kl  13 08 24

**Pull request merged**